cpp/src/parquet/page_index.h

```diff
@@ -186,8 +191,7 @@ class PARQUET_EXPORT PageIndexReader {
   /// that creates this PageIndexReader.
   static std::shared_ptr<PageIndexReader> Make(
       ::arrow::io::RandomAccessFile* input, std::shared_ptr<FileMetaData> file_metadata,
-      const ReaderProperties& properties,
-      std::shared_ptr<InternalFileDecryptor> file_decryptor = NULLPTR);
+      const ReaderProperties& properties, InternalFileDecryptor* file_decryptor);
```

Comment:
Why do some APIs take a raw pointer and others a shared_ptr? Is there a reason for the inconsistency?

Reply:
For `RandomAccessFile*` and `InternalFileDecryptor*`, I want to explicitly state that returned PageIndexReader does not own these objects and they are owned by the parquet file reader. However, it is unfortunate that no one owns `file_metadata` so it must be a `std::shared_ptr<FileMetaData>`.

Reply:
My question is more about `std::shared_ptr<Decryptor>` vs. `InternalFileDecryptor*`.

Reply:
OK. `std::shared_ptr<Decryptor>` is due to the expectation of `DeserializeMessage` call.